jax/_src/lax/lax.py

```diff
@@ -2940,18 +2975,38 @@ def _conv_general_dilated_translation_rule(
     # error bound (e.g. 1p-24 in case of float32) would be relative to the
     # maximum of real and imaginary parts of the result instead of being
     # satisfied by the real and imaginary parts independently of each other.
+    if preferred_element_type is not None:
+      # Convert complex dtype to types used for real and imaginary parts
+      assert np.issubdtype(preferred_element_type, np.complexfloating)
+      preferred_element_type = xla_client.dtype_to_etype(
+          np.dtype(f"f{np.dtype(preferred_element_type).itemsize // 2}"))
```

Comment:
Not sure if there is a nicer way of doing this. I'm very open to suggestions :)

Reply:
I'd use a helper function, similar to this: https://github.com/google/jax/blob/dbe9ff5f2657f9c124f60e4183d05423d913af8a/jaxlib/rocsolver.py#L36-L47

Reply:
or, since you know this is complex and there are only two possibilities, an if/else statement would probably suffice :grin:

Reply:
Great idea! I wasn't aware that there are only two complex types. Fixed in c8f4173b6ec3093d8a3130a26fe31a0e764dc963